core/metacling/src/TClingCallFunc.cxx

```diff
@@ -246,29 +177,35 @@ namespace {
    static
    long long sv_to_long_long(const cling::Value &val)
    {
-      return sv_to<long long>(val);
+      return val.getAs<long long>();
```

Comment:
Without explicit casting (rather than reinterpret cast like here), the value will be garbled on big endian platforms.

Reply:
Hm, are you sure about this? Is there a platform where we can check this?

Or are you simply suggesting `return (long long) val.getAs<long long>()`?

Reply:
> Hm, are you sure about this?

Yes.

> Is there a platform where we can check this?

Big endian is almost dead; some Power machines can be booted in big endian mode. You can try here: https://www.godbolt.org/z/zzb4Mx5d9

> Or are you simply suggesting `return (long long) val.getAs<long long>()`?

No, not good enough. You need to access the value as it was stored (`getAs<the-type-as-written-into-the-cling-Value>()`) and then cast that.

Reply:
I think I understand what you say now. I think here we are save as we do the recast to the concrete type: we have constructs like `case BuiltinType::Bool: vh.u.b = (bool) sv_to_ulong_long(fArgVals[i]);`

Using the link you sent I discovered something that's important for the further refactoring though.

```cpp
    int i = 17;
    char c = *(char*)&i;
    c= (char) i; // === `c = i;` === `c = (unsigned long long) i;`
    return c;
```

It seems that we can re-cast to the longest type as long as the underlying representation is the expected one. So we don't need to enumerate all types but just enumerate integral (or enum), real, etc.

Reply:
Wouldn't that depend on the endianess?

Reply:
I see, we don't store integers of different sizes, so endianess is indeed not an issue:
https://github.com/root-project/root/blob/c2732039eb37cf2fa21f227353aae7e16d3bb935/core/metacling/src/TClingCallFunc.cxx#L2325-L2330


Still, `val.getAs<long long>` on a `val` that doesn't store a `long long` (but say a `double`) can give a wrong value. Unless it's guaranteed that the `Value` stores the same type as being read here, I'd be much happier if we were to use `simplisticCastAs` rather than `getAs`.

Reply:
The simplisticCast comes in #11193. Should I bring the commit here.

Reply:
Added in this PR.